### DISCUSSION TITLE
Is keycloak affected by CVE-2024-21170

### DISCUSSION THREAD

**Discussion by jsto** [keycloak/keycloak#32647] [Q&A] [answered]
Keycloak has mysql-connector-j with version 8.3.0 whereas the recommendation is MySQL Connectors(Connector/Python) ≤ 8.4.0

**Comment by haloblaks** [comment, marked as answer]
Hi. This came up from our scan as well. Looks like the same version is used in 25.0.6
https://github.com/keycloak/keycloak/blob/ab33487eb861829395214506c14928ed13342a2e/pom.xml#L159

**Reply by shawkins** [reply]
The CVE is specifically against the python connector, not the java one. Keycloak is unaffected.

Also note that we generally align this dependency with Quarkus - if there were an outstanding CVE like this it should be addressed in Quarkus, then picked up in Keycloak.

**Reply by haloblaks** [reply]
Hi. Thank you. We'll mark this as FP then.

**Reply by codespearhead** [reply]
@jsto Can you [mark shawkins's comment as the answer](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) so it doesn't show up as unanswered in the GitHub discussions tab?
